### PR TITLE
Add soname to shared library

### PR DESCRIPTION
**Pull request by bkmgit**
Update CMakelists to add soname to shared library
using  VERSION ${F3D_VERSION} and SOVERSION ${F3D_MAJOR_VERSION}

https://github.com/f3d-app/f3d/issues/821

### CODE REVIEW

**Review by Meakk — APPROVED** [review]
Looks good to me. Wdyt @mwestphal?

**Comment by bkmgit** [comment]
Ubuntu failures can probably be fixed by adding libbsd-dev to the image, or using apt to install this during the CI build

**Comment by mwestphal** [comment]
Actually caused by a slight change in our docker setup, will be fixed soon

**Comment by mwestphal** [comment]
Ill need to check and understand how sonames work

**Comment by bkmgit** [comment]
Maybe helpful:
https://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html
https://en.wikipedia.org/wiki/Soname
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/developer_guide/creating-libraries-gcc

**Comment by mwestphal** [comment]
Thanks for the links !

**Comment by Meakk** [comment]
Note that I don't see the library with the version suffix or the symlinks in the CI logs here: https://github.com/f3d-app/f3d/actions/runs/5037504197/jobs/9034730385?pr=822#step:14:123
Please double check it's properly generated on your side.

**Comment by Meakk** [comment]
Thank you. It looks like it is what we are expecting 👍

**Comment by mwestphal** [comment]
@mzf-guest You may want to take a look at this.

File: library/CMakeLists.txt [hunk 1]
```diff
@@ -108,6 +108,8 @@ set_target_properties(libf3d PROPERTIES
   POSITION_INDEPENDENT_CODE ON
   OUTPUT_NAME "f3d"
   PDB_NAME "libf3d"
+  VERSION ${F3D_VERSION}
```
[the remark this answers is not in the document]
Comment (bkmgit) [reply library/CMakeLists.txt]: Updated.

**Review by mwestphal — APPROVED** [review]

**Comment by mwestphal** [comment]
Okay for you @Meakk ?

**Comment by Meakk** [comment]
Yes

**Review by Meakk — APPROVED** [review]

**Pull request merged**